cmake/adjust/template/vendor/impl/Clang.cmake

```diff
@@ -0,0 +1,89 @@
+include("${CMAKE_CURRENT_LIST_DIR}/reset.cmake")
+
+# vendor template with options fitting for both C and CXX LANGs
+
+if(NOT DEFINED LANG)
+	message(FATAL_ERROR "LANG must be defined!")
+endif()
+
+if(NBL_REQUEST_SSE_4_2)
+	NBL_REQUEST_COMPILE_OPTION_SUPPORT(LANG ${LANG} OPTIONS
+		-msse4.2 # https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang1-msse4.2
+	)
+endif()
+
+if(NBL_REQUEST_SSE_AVX2)
+	NBL_REQUEST_COMPILE_OPTION_SUPPORT(LANG ${LANG} OPTIONS
+		-mavx2 # https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-mavx2
+	)
+endif()
+
+NBL_REQUEST_COMPILE_OPTION_SUPPORT(LANG ${LANG} OPTIONS
+	# latest Clang(CL) 19.1.1 shipped with VS seems to require explicitly features to be listed (simdjson)
+	# TODO: Yas, use with REQUEST_VAR, if the request fail then do not promote simdjson to build with 
+	# HASWELL implementation because those flags + avx2 compose subset it wants in this case
+
+	# also instead of enabling single options maybe we could consider requesting an
+	# instruction implementation set instead, eg -march=haswel, though this approach
+	# could add a few more flags then we actually need while building - to rethink
```

Comment:
@devshgraphicsprogramming

Reply:
https://github.com/llvm/llvm-project/blob/b46f16c8d7f4a9e37271cdc4717dd858a4c813dc/llvm/lib/Target/X86/X86.td#L937

Reply:
`-march` is a bad idea unless you're deploying on a particular CPU (i.e. renderfarm or amazon AWS)

Reply:
> `-march` is a bad idea unless you're deploying on a particular CPU (i.e. renderfarm or amazon AWS)

understood, also found it too much - resolved